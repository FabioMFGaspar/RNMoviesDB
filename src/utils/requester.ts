import {Dispatch} from 'redux';
import {searchMovieByQuery, fetchMovieDetails} from '../libraries/movieDb';
import {
  setSearchMovies,
  setMovieDetails,
  clearMovieDetails,
} from '../store/actions';
import {movieFormatter, detailsFormatter} from './formatter';

// Maybe this isn't the best name for the file...
// This functions are responsible for getting information from
// the MovieDB library, formatting it and dispatching it to the store
// be it the movies list in the search bar or the movie details for the
// details page
export const findMovieByQuery = async ({
  query,
  dispatch,
}: {
  query: string;
  dispatch: Dispatch;
}) => {
  const {results} = await searchMovieByQuery(query);

  if (results) {
    const formattedMovies = movieFormatter(results);
    dispatch(setSearchMovies(formattedMovies));
  } else {
    dispatch(setSearchMovies([]));
  }
};

export const getMovieById = async ({
  movieId,
  dispatch,
}: {
  movieId: number;
  dispatch: Dispatch;
}) => {
  const result = await fetchMovieDetails(movieId);

  if (Object.keys(result).length) {
    const formattedDetails = detailsFormatter(result);
    dispatch(setMovieDetails(formattedDetails));
  } else {
    dispatch(clearMovieDetails());
  }
};
