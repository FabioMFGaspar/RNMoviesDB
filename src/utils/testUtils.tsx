import React, {ReactElement} from 'react';
import {createStore} from 'redux';
import {Provider} from 'react-redux';
import {render} from '@testing-library/react-native';
import reducers, {initialState} from '../store/reducers';
import {StoreState} from '../types/store';

// This component is used to test the pages and make sure they
// are connected to the redux store
export function renderWithRedux(
  component: ReactElement,
  state: StoreState = initialState,
) {
  const store = createStore(reducers, state);
  const queries = render(<Provider store={store}>{component}</Provider>);

  return {
    ...queries,
    store,
  };
}
