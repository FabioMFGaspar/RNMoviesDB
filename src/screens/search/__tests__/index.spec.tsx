import React from 'react';
import {fireEvent} from '@testing-library/react-native';
import {renderWithRedux} from '../../../utils/testUtils';
import {findMovieByQuery, getMovieById} from '../../../utils/requester';
import Search from '..';
import {mockMovies} from '../../../mocks/movieDB.mock';

jest.mock('@react-navigation/native');
jest.mock('../../../utils/requester', () => ({
  findMovieByQuery: jest.fn(),
  getMovieById: jest.fn(),
}));


describe('Search Screen test suite', () => {
  it('Search snapshot test', () => {
    const {toJSON} = renderWithRedux(<Search />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('Snapshot with movie list', () => {
    const {toJSON} = renderWithRedux(<Search />, {
      searchMovies: mockMovies,
      movieDetails: {},
    });
    expect(toJSON()).toMatchSnapshot();
  });

  it('Should find the movie in the list', () => {
    const {queryAllByText, queryByText} = renderWithRedux(<Search />, {
      searchMovies: mockMovies,
      movieDetails: {},
    });

    const movie = queryAllByText('The Avengers');
    const noMovie = queryByText('No movies found...');
    const searchForMovie = queryByText('Search for a movie...');

    expect(movie).toBeTruthy();
    expect(noMovie).toBeNull();
    expect(searchForMovie).toBeNull();
  });

  it('Should find the empty component', () => {
    const {queryByText} = renderWithRedux(<Search />, {
      searchMovies: [],
      movieDetails: {},
    });

    const noMovie = queryByText('No movies found...');
    const searchForMovie = queryByText('Search for a movie...');

    expect(noMovie).toBeNull();
    expect(searchForMovie).toBeTruthy();
  });

  it('Should trigger onChangeText', () => {
    const mockHandle = jest.fn();
    // @ts-ignore
    findMovieByQuery.mockImplementation(mockHandle);
    const {getByPlaceholderText} = renderWithRedux(<Search />, {
      searchMovies: mockMovies,
      movieDetails: {},
    });

    fireEvent(getByPlaceholderText('Search...'), 'onChangeText', 'dummyChange');
    expect(mockHandle).toHaveBeenCalled();
  });

  it('Should trigger movie onPress', () => {
    const mockHandle = jest.fn();
    // @ts-ignore
    getMovieById.mockImplementation(mockHandle);
    const {getByText} = renderWithRedux(<Search />, {
      searchMovies: mockMovies,
      movieDetails: {},
    });

    const movie = getByText('Avengers: Age of Ultron');
    fireEvent.press(movie);

    expect(mockHandle).toHaveBeenCalled();
  });
});
