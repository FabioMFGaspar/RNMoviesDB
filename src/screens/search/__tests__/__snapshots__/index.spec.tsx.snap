// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Screen test suite Search snapshot test 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#000033",
      "display": "flex",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#000033",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#212145",
          "borderRadius": 5,
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginHorizontal": 15,
          "marginVertical": 10,
          "padding": 10,
        }
      }
    >
      <TextInput
        onChangeText={[Function]}
        placeholder="Search..."
        placeholderTextColor="white"
        style={
          Object {
            "backgroundColor": "#212145",
            "color": "white",
            "flex": 1,
            "padding": 0,
          }
        }
        value=""
      />
      <Image
        source={
          Object {
            "uri": "https://img.icons8.com/ios-glyphs/30/000000/search--v2.png",
          }
        }
        style={
          Object {
            "height": 25,
            "marginLeft": 5,
            "tintColor": "white",
            "width": 25,
          }
        }
      />
    </View>
  </View>
  <RCTScrollView
    ItemSeparatorComponent={[Function]}
    ListEmptyComponent={[Function]}
    data={Array []}
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    viewabilityConfigCallbackPairs={Array []}
  >
    <View>
      <View
        style={
          Object {
            "padding": 15,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 14,
            }
          }
        >
          Search for a movie...
        </Text>
      </View>
    </View>
  </RCTScrollView>
</View>
`;

exports[`Search Screen test suite Snapshot with movie list 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#000033",
      "display": "flex",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#000033",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#212145",
          "borderRadius": 5,
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginHorizontal": 15,
          "marginVertical": 10,
          "padding": 10,
        }
      }
    >
      <TextInput
        onChangeText={[Function]}
        placeholder="Search..."
        placeholderTextColor="white"
        style={
          Object {
            "backgroundColor": "#212145",
            "color": "white",
            "flex": 1,
            "padding": 0,
          }
        }
        value=""
      />
      <Image
        source={
          Object {
            "uri": "https://img.icons8.com/ios-glyphs/30/000000/search--v2.png",
          }
        }
        style={
          Object {
            "height": 25,
            "marginLeft": 5,
            "tintColor": "white",
            "width": 25,
          }
        }
      />
    </View>
  </View>
  <RCTScrollView
    ItemSeparatorComponent={[Function]}
    ListEmptyComponent={[Function]}
    data={
      Array [
        Object {
          "backImgPath": "https://image.tmdb.org/t/p/w400/hbn46fQaRmlpBuUrEiFqv0GDL6Y.jpg",
          "id": 24428,
          "overview": "When an unexpected enemy emerges and threatens global safety and security, Nick Fury, director of the international peacekeeping agency known as S.H.I.E.L.D., finds himself in need of a team to pull the world back from the brink of disaster. Spanning the globe, a daring recruitment effort begins!",
          "title": "The Avengers",
        },
        Object {
          "backImgPath": "https://image.tmdb.org/t/p/w400/8YW4rwWQgC2JRlBcpStMNUko13k.jpg",
          "id": 9320,
          "overview": "British Ministry agent John Steed, under direction from \\"Mother\\", investigates a diabolical plot by arch-villain Sir August de Wynter to rule the world with his weather control machine. Steed investigates the beautiful Doctor Mrs. Emma Peel, the only suspect, but simultaneously falls for her and joins forces with her to combat Sir August.",
          "title": "The Avengers",
        },
        Object {
          "backImgPath": "https://image.tmdb.org/t/p/w400/570qhjGZmGPrBGnfx70jcwIuBr4.jpg",
          "id": 99861,
          "overview": "When Tony Stark tries to jumpstart a dormant peacekeeping program, things go awry and Earthâ€™s Mightiest Heroes are put to the ultimate test as the fate of the planet hangs in the balance. As the villainous Ultron emerges, it is up to The Avengers to stop him from enacting his terrible plans, and soon uneasy alliances and unexpected action pave the way for an epic and unique global adventure.",
          "title": "Avengers: Age of Ultron",
        },
        Object {
          "backImgPath": "https://image.tmdb.org/t/p/w400null",
          "id": 392031,
          "overview": "This Australian children's film is about scientist Bill Stewart goes to Fiji with his son Tim to investigate the appearance of the Crown of Thorns starfish in the reefs off the island.",
          "title": "Avengers of the Reef",
        },
        Object {
          "backImgPath": "https://image.tmdb.org/t/p/w400/85NqI4WuCim6dZexmTPUAi13Af0.jpg",
          "id": 14611,
          "overview": "Mysterious Wakanda lies in the darkest heart of Africa, unknown to most of the world. An isolated land hidden behind closed borders, fiercely protected by its young king - the Black Panther. But when brutal alien invaders attack, the threat leaves the Black Panther with no option but to go against the sacred decrees of his people and ask for help from outsiders.",
          "title": "Ultimate Avengers 2",
        },
        Object {
          "backImgPath": "https://image.tmdb.org/t/p/w400/mZO4V0ALx15QTgWr4SaXYGT7i60.jpg",
          "id": 14609,
          "overview": "When a nuclear missile was fired at Washington in 1945, Captain America managed to detonate it in the upper atmosphere. But then he fell miles into the icy depths of the North Atlantic, where he remained lost for over sixty years. But now, with the world facing the very same evil, Captain America must rise again as our last hope for survival.",
          "title": "Ultimate Avengers",
        },
        Object {
          "backImgPath": "https://image.tmdb.org/t/p/w400/6ajw8PjpnelE6l28VZRB6PX4KM.jpg",
          "id": 65591,
          "overview": "A brilliant continuation of the saga, now in the city. The target of young patriots is a map held in the safe locker in the headquarters of secret police!",
          "title": "The New Adventures of the Elusive Avengers",
        },
        Object {
          "backImgPath": "https://image.tmdb.org/t/p/w400/cx3GroQGuwJGYfhD13iLyNMV9X8.jpg",
          "id": 109088,
          "overview": "The famous story of the Shaolin Temple's betrayal by the White-Browed Hermit, and the subsequent revenge by Shaolin firebrand Fang Shih-yu, is the stuff of legend. It has been filmed many times by many directors, but few are remembered as fondly as this production. The potent combination of director Chang Cheh and international idol Alexander Fu Sheng caught lightning in a lens.",
          "title": "The Shaolin Avengers",
        },
        Object {
          "backImgPath": "https://image.tmdb.org/t/p/w400/cd3VcJBzLJs3gjX0LpE1ZL7hHqs.jpg",
          "id": 64128,
          "overview": "Horse race tipster and journalist Metcalfe is picked for the job of foreign correspondent in Norway when Hitler invades Poland. On the way to Norway his boat is attacked by a German U-Boat, however when he tells the navy about it they disbelief him and, to make matters worse, he is removed from his job. When German forces invade Norway, Metcalfe returns determined to uncover what is going on and stop the Germans in their tracks.",
          "title": "The Day Will Dawn",
        },
        Object {
          "backImgPath": "https://image.tmdb.org/t/p/w400null",
          "id": 275663,
          "overview": "When two acrobats are fired for fighting with punks in the audience, they go to live with an aunt who's being pressured to sell her house for a real estate development. The developer's nasty son, Lee Fu, decides to muscle the sale, and soon he's at war with the acrobats, plus their unlikely ally, an American named John who used to be Lee Fu's friend. The acrobats open a kung fu school, the scene of several battles with Lee Fu's thugs. A fight to the death, jail time, auntie's surprise decision, a budding acting career, a possessive girlfriend, a debilitating injury, a friendship that needs recalibrating, and Lee Fu's avenger are all in the mix before the end.",
          "title": "The Lama Avenger",
        },
        Object {
          "backImgPath": "https://image.tmdb.org/t/p/w400null",
          "id": 377364,
          "overview": "An insider's look at the first year of an activist group known as the Lesbian Avengers.",
          "title": "Lesbian Avengers Eat Fire Too",
        },
        Object {
          "backImgPath": "https://image.tmdb.org/t/p/w400/5nwuBYksiGkrACCVgq086L9zwWm.jpg",
          "id": 40081,
          "overview": "Crippled Avengers is a 1978 Shaw Brothers kung fu film directed by Chang Cheh and starring four members of the Venom Mob. It has been released in North America as Mortal Combat and Return of the 5 Deadly Venoms. The film follows a group of martial artists seeking revenge after being crippled by Tu Tin-To (Chen Kuan Tai), a martial arts master, and his son (Lu Feng).",
          "title": "Crippled Avengers",
        },
        Object {
          "backImgPath": "https://image.tmdb.org/t/p/w400null",
          "id": 187745,
          "overview": "Hercules, having agreed to restore justice to Mycenae, confronts evil Prince Myles. The villain is invested with power over giant bronze warriors by his mother PasiphaÃ«, a sorceress in Hades. Myles assassinates the king and abducts his cousin, Queen Ate, through whom he plans to inherit the throne. Although Zeus temporarily removes Hercules' strength for killing Eurystheus, an innocent man framed by Myles, the god restores his powers. Hercules then saves Ate and destroys Myles, PasiphaÃ«, and the giant warriors.",
          "title": "Hercules vs. the Giant Warriors",
        },
        Object {
          "backImgPath": "https://image.tmdb.org/t/p/w400/yeKT2gNFxHGbTT3Htj5PE9IerGJ.jpg",
          "id": 161097,
          "overview": "Feature-length documentary about the rise of Marvel Studios and their films leading up to, and including, The Avengers.",
          "title": "Marvel Studios: Building a Cinematic Universe",
        },
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    viewabilityConfigCallbackPairs={Array []}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#2b2b3b",
              "borderRadius": 10,
              "display": "flex",
              "marginHorizontal": 15,
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "https://image.tmdb.org/t/p/w400/hbn46fQaRmlpBuUrEiFqv0GDL6Y.jpg",
              }
            }
            style={
              Object {
                "borderTopLeftRadius": 10,
                "borderTopRightRadius": 10,
                "height": 200,
                "width": 360,
              }
            }
          />
          <View
            style={
              Object {
                "padding": 10,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#e6e6ff",
                  "fontSize": 16,
                  "fontWeight": "bold",
                  "marginBottom": 5,
                }
              }
            >
              The Avengers
            </Text>
            <Text
              numberOfLines={3}
              style={
                Object {
                  "color": "#8c8c8c",
                  "fontSize": 12,
                }
              }
            >
              When an unexpected enemy emerges and threatens global safety and security, Nick Fury, director of the international peacekeeping agency known as S.H.I.E.L.D., finds himself in need of a team to pull the world back from the brink of disaster. Spanning the globe, a daring recruitment effort begins!
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "margin": 5,
            }
          }
        />
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#2b2b3b",
              "borderRadius": 10,
              "display": "flex",
              "marginHorizontal": 15,
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "https://image.tmdb.org/t/p/w400/8YW4rwWQgC2JRlBcpStMNUko13k.jpg",
              }
            }
            style={
              Object {
                "borderTopLeftRadius": 10,
                "borderTopRightRadius": 10,
                "height": 200,
                "width": 360,
              }
            }
          />
          <View
            style={
              Object {
                "padding": 10,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#e6e6ff",
                  "fontSize": 16,
                  "fontWeight": "bold",
                  "marginBottom": 5,
                }
              }
            >
              The Avengers
            </Text>
            <Text
              numberOfLines={3}
              style={
                Object {
                  "color": "#8c8c8c",
                  "fontSize": 12,
                }
              }
            >
              British Ministry agent John Steed, under direction from "Mother", investigates a diabolical plot by arch-villain Sir August de Wynter to rule the world with his weather control machine. Steed investigates the beautiful Doctor Mrs. Emma Peel, the only suspect, but simultaneously falls for her and joins forces with her to combat Sir August.
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "margin": 5,
            }
          }
        />
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#2b2b3b",
              "borderRadius": 10,
              "display": "flex",
              "marginHorizontal": 15,
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "https://image.tmdb.org/t/p/w400/570qhjGZmGPrBGnfx70jcwIuBr4.jpg",
              }
            }
            style={
              Object {
                "borderTopLeftRadius": 10,
                "borderTopRightRadius": 10,
                "height": 200,
                "width": 360,
              }
            }
          />
          <View
            style={
              Object {
                "padding": 10,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#e6e6ff",
                  "fontSize": 16,
                  "fontWeight": "bold",
                  "marginBottom": 5,
                }
              }
            >
              Avengers: Age of Ultron
            </Text>
            <Text
              numberOfLines={3}
              style={
                Object {
                  "color": "#8c8c8c",
                  "fontSize": 12,
                }
              }
            >
              When Tony Stark tries to jumpstart a dormant peacekeeping program, things go awry and Earthâ€™s Mightiest Heroes are put to the ultimate test as the fate of the planet hangs in the balance. As the villainous Ultron emerges, it is up to The Avengers to stop him from enacting his terrible plans, and soon uneasy alliances and unexpected action pave the way for an epic and unique global adventure.
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "margin": 5,
            }
          }
        />
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#2b2b3b",
              "borderRadius": 10,
              "display": "flex",
              "marginHorizontal": 15,
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "https://image.tmdb.org/t/p/w400null",
              }
            }
            style={
              Object {
                "borderTopLeftRadius": 10,
                "borderTopRightRadius": 10,
                "height": 200,
                "width": 360,
              }
            }
          />
          <View
            style={
              Object {
                "padding": 10,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#e6e6ff",
                  "fontSize": 16,
                  "fontWeight": "bold",
                  "marginBottom": 5,
                }
              }
            >
              Avengers of the Reef
            </Text>
            <Text
              numberOfLines={3}
              style={
                Object {
                  "color": "#8c8c8c",
                  "fontSize": 12,
                }
              }
            >
              This Australian children's film is about scientist Bill Stewart goes to Fiji with his son Tim to investigate the appearance of the Crown of Thorns starfish in the reefs off the island.
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "margin": 5,
            }
          }
        />
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#2b2b3b",
              "borderRadius": 10,
              "display": "flex",
              "marginHorizontal": 15,
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "https://image.tmdb.org/t/p/w400/85NqI4WuCim6dZexmTPUAi13Af0.jpg",
              }
            }
            style={
              Object {
                "borderTopLeftRadius": 10,
                "borderTopRightRadius": 10,
                "height": 200,
                "width": 360,
              }
            }
          />
          <View
            style={
              Object {
                "padding": 10,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#e6e6ff",
                  "fontSize": 16,
                  "fontWeight": "bold",
                  "marginBottom": 5,
                }
              }
            >
              Ultimate Avengers 2
            </Text>
            <Text
              numberOfLines={3}
              style={
                Object {
                  "color": "#8c8c8c",
                  "fontSize": 12,
                }
              }
            >
              Mysterious Wakanda lies in the darkest heart of Africa, unknown to most of the world. An isolated land hidden behind closed borders, fiercely protected by its young king - the Black Panther. But when brutal alien invaders attack, the threat leaves the Black Panther with no option but to go against the sacred decrees of his people and ask for help from outsiders.
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "margin": 5,
            }
          }
        />
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#2b2b3b",
              "borderRadius": 10,
              "display": "flex",
              "marginHorizontal": 15,
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "https://image.tmdb.org/t/p/w400/mZO4V0ALx15QTgWr4SaXYGT7i60.jpg",
              }
            }
            style={
              Object {
                "borderTopLeftRadius": 10,
                "borderTopRightRadius": 10,
                "height": 200,
                "width": 360,
              }
            }
          />
          <View
            style={
              Object {
                "padding": 10,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#e6e6ff",
                  "fontSize": 16,
                  "fontWeight": "bold",
                  "marginBottom": 5,
                }
              }
            >
              Ultimate Avengers
            </Text>
            <Text
              numberOfLines={3}
              style={
                Object {
                  "color": "#8c8c8c",
                  "fontSize": 12,
                }
              }
            >
              When a nuclear missile was fired at Washington in 1945, Captain America managed to detonate it in the upper atmosphere. But then he fell miles into the icy depths of the North Atlantic, where he remained lost for over sixty years. But now, with the world facing the very same evil, Captain America must rise again as our last hope for survival.
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "margin": 5,
            }
          }
        />
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#2b2b3b",
              "borderRadius": 10,
              "display": "flex",
              "marginHorizontal": 15,
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "https://image.tmdb.org/t/p/w400/6ajw8PjpnelE6l28VZRB6PX4KM.jpg",
              }
            }
            style={
              Object {
                "borderTopLeftRadius": 10,
                "borderTopRightRadius": 10,
                "height": 200,
                "width": 360,
              }
            }
          />
          <View
            style={
              Object {
                "padding": 10,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#e6e6ff",
                  "fontSize": 16,
                  "fontWeight": "bold",
                  "marginBottom": 5,
                }
              }
            >
              The New Adventures of the Elusive Avengers
            </Text>
            <Text
              numberOfLines={3}
              style={
                Object {
                  "color": "#8c8c8c",
                  "fontSize": 12,
                }
              }
            >
              A brilliant continuation of the saga, now in the city. The target of young patriots is a map held in the safe locker in the headquarters of secret police!
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "margin": 5,
            }
          }
        />
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#2b2b3b",
              "borderRadius": 10,
              "display": "flex",
              "marginHorizontal": 15,
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "https://image.tmdb.org/t/p/w400/cx3GroQGuwJGYfhD13iLyNMV9X8.jpg",
              }
            }
            style={
              Object {
                "borderTopLeftRadius": 10,
                "borderTopRightRadius": 10,
                "height": 200,
                "width": 360,
              }
            }
          />
          <View
            style={
              Object {
                "padding": 10,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#e6e6ff",
                  "fontSize": 16,
                  "fontWeight": "bold",
                  "marginBottom": 5,
                }
              }
            >
              The Shaolin Avengers
            </Text>
            <Text
              numberOfLines={3}
              style={
                Object {
                  "color": "#8c8c8c",
                  "fontSize": 12,
                }
              }
            >
              The famous story of the Shaolin Temple's betrayal by the White-Browed Hermit, and the subsequent revenge by Shaolin firebrand Fang Shih-yu, is the stuff of legend. It has been filmed many times by many directors, but few are remembered as fondly as this production. The potent combination of director Chang Cheh and international idol Alexander Fu Sheng caught lightning in a lens.
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "margin": 5,
            }
          }
        />
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#2b2b3b",
              "borderRadius": 10,
              "display": "flex",
              "marginHorizontal": 15,
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "https://image.tmdb.org/t/p/w400/cd3VcJBzLJs3gjX0LpE1ZL7hHqs.jpg",
              }
            }
            style={
              Object {
                "borderTopLeftRadius": 10,
                "borderTopRightRadius": 10,
                "height": 200,
                "width": 360,
              }
            }
          />
          <View
            style={
              Object {
                "padding": 10,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#e6e6ff",
                  "fontSize": 16,
                  "fontWeight": "bold",
                  "marginBottom": 5,
                }
              }
            >
              The Day Will Dawn
            </Text>
            <Text
              numberOfLines={3}
              style={
                Object {
                  "color": "#8c8c8c",
                  "fontSize": 12,
                }
              }
            >
              Horse race tipster and journalist Metcalfe is picked for the job of foreign correspondent in Norway when Hitler invades Poland. On the way to Norway his boat is attacked by a German U-Boat, however when he tells the navy about it they disbelief him and, to make matters worse, he is removed from his job. When German forces invade Norway, Metcalfe returns determined to uncover what is going on and stop the Germans in their tracks.
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "margin": 5,
            }
          }
        />
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#2b2b3b",
              "borderRadius": 10,
              "display": "flex",
              "marginHorizontal": 15,
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "https://image.tmdb.org/t/p/w400null",
              }
            }
            style={
              Object {
                "borderTopLeftRadius": 10,
                "borderTopRightRadius": 10,
                "height": 200,
                "width": 360,
              }
            }
          />
          <View
            style={
              Object {
                "padding": 10,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#e6e6ff",
                  "fontSize": 16,
                  "fontWeight": "bold",
                  "marginBottom": 5,
                }
              }
            >
              The Lama Avenger
            </Text>
            <Text
              numberOfLines={3}
              style={
                Object {
                  "color": "#8c8c8c",
                  "fontSize": 12,
                }
              }
            >
              When two acrobats are fired for fighting with punks in the audience, they go to live with an aunt who's being pressured to sell her house for a real estate development. The developer's nasty son, Lee Fu, decides to muscle the sale, and soon he's at war with the acrobats, plus their unlikely ally, an American named John who used to be Lee Fu's friend. The acrobats open a kung fu school, the scene of several battles with Lee Fu's thugs. A fight to the death, jail time, auntie's surprise decision, a budding acting career, a possessive girlfriend, a debilitating injury, a friendship that needs recalibrating, and Lee Fu's avenger are all in the mix before the end.
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "margin": 5,
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "height": 0,
          }
        }
      />
    </View>
  </RCTScrollView>
</View>
`;
